JPS_VIRTUALSENSOR/AermodAgent/src/main/java/com/cmclinnovations/aermod/AermodAgent.java

```diff
@@ -177,82 +181,136 @@ public void processRequestParameters(DerivationInputs derivationInputs, Derivati
         String sourceLayerName = "source_layer";
         String elevationLayerName = "elevation_layer";
+        // create geoserver layer based for that
+        GeoServerClient geoServerClient = GeoServerClient.getInstance();
+        // make sure style is uploaded first
+        uploadStyle(geoServerClient);
+        // then create a layer with that style as default
+        GeoServerVectorSettings geoServerVectorSettings = new GeoServerVectorSettings();
+        geoServerVectorSettings.setDefaultStyle("dispersion_style");
 
-        // Get contour plots as geoJSON objects from PythonService and upload them to
-        // PostGIS using GDAL
-
+        // Upload point sources layer to POSTGIS and GeoServer
         JSONObject pointSourceFeatures = aermod.createStaticPointSourcesJSON(staticPointSources);
-
         GDALClient gdalClient = GDALClient.getInstance();
-
         gdalClient.uploadVectorStringToPostGIS(EnvConfig.DATABASE, EnvConfig.SOURCE_LAYER,
                 pointSourceFeatures.toString(),
                 new Ogr2OgrOptions(), true);
+        geoServerClient.createPostGISLayer(EnvConfig.GEOSERVER_WORKSPACE, EnvConfig.DATABASE, EnvConfig.SOURCE_LAYER,
+                new GeoServerVectorSettings());
 
-        JSONObject geoJSON2 = aermod.getGeoJSON(EnvConfig.PYTHON_SERVICE_ELEVATION_URL, outFileURL, srid, 0.0);
+        // Upload elevation contour plot to POSTGIS and GeoServer
+        // The receptor.dat file may have been previously created by running AERMAP. If
+        // so, it should not be overwritten.
+        if (Files.notExists(simulationDirectory.resolve("aermod").resolve("receptor.dat")))
+            aermod.createAERMODReceptorInput(scope, nx, ny, srid);
+        String outFileURL = aermod.uploadToFileServer("receptor.dat");
+        JSONObject geoJSON2 = aermod.getGeoJSON(EnvConfig.PYTHON_SERVICE_ELEVATION_URL, outFileURL, srid, 0.0, null);
         gdalClient.uploadVectorStringToPostGIS(EnvConfig.DATABASE, elevationLayerName, geoJSON2.toString(),
                 new Ogr2OgrOptions(), true);
+        geoServerClient.createPostGISLayer(EnvConfig.GEOSERVER_WORKSPACE, EnvConfig.DATABASE, elevationLayerName,
+                geoServerVectorSettings);
 
-        for (int i = 0; i < dispLayerNames.size(); i++) {
-            double height = receptorHeights.get(i);
-            JSONObject geoJSON = aermod.getGeoJSON(EnvConfig.PYTHON_SERVICE_URL, outputFileURL, srid, height);
-            String dispLayerName = dispLayerNames.get(i);
-            gdalClient.uploadVectorStringToPostGIS(EnvConfig.DATABASE, dispLayerName, geoJSON.toString(),
-                    new Ogr2OgrOptions(), true);
-        }
+        for (String pollutId : pollutantsList) {
+            // create emissions input
+            if (aermod.createPointsFile(allSources, srid, pollutId) != 0) {
+                LOGGER.error("Failed to create points emissions file for pollutant {} at timestep {}", pollutId,
+                        simulationTime);
```

Comment:
I'd not log an "error" for a pollutant skipped due to the maxFlowRate check, it is misleading. Ideally different log messages for different outcomes